src/subsplit_dag_storage.hpp

```diff
@@ -575,6 +599,40 @@ class SubsplitDAGStorage {
     vertices_.ResetHost(&host.vertices_);
   }
 
+  void ConnectVertices(LineId lineId) {
+    auto& line = lines_.at(lineId);
+    auto& parent = vertices_.at(line.GetParent());
+    auto& child = vertices_.at(line.GetChild());
+    parent.AddNeighbor(Direction::Leafward, line.GetClade(), child.GetId(), lineId);
+    child.AddNeighbor(Direction::Rootward, line.GetClade(), parent.GetId(), lineId);
+  }
+
+  void ConnectAllVertices() {
+    for (size_t i = 0; i < vertices_.size(); ++i) {
+      vertices_[i].ClearNeighbors();
+    }
+    for (size_t i = 0; i < lines_.size(); ++i) {
+      if (lines_[i].GetId() == NoId) {
+        continue;
+      }
+      ConnectVertices(lines_[i].GetId());
+    }
+  }
+
+  std::optional<std::reference_wrapper<const DAGVertex>> FindRoot() const {
```

Comment:
Ordering of nodes is a concept of the SubsplitDAG, not the storage layer. There are situations (like in Graft, or when adding nodes) when the ordering is yet to be enforced. For users of SubsplitDAG ordering looks like an atomic operation - there is no call in the sDAG interface that would leave it in a state where the root isn't the last node, but during internal usage we don't have such guarantee. The function is called FindRoot instead of GetRoot to denote higher complexity than O(1).